lid_mismatches.py

```diff
@@ -0,0 +1,166 @@
+        type=str,
+        required=True,
+        help="CSV file with language identification results.",
+    )
+    parser.add_argument(
+        "--tokenized_csv_filename",
+        type=str,
+        required=True,
+        help="CSV file with tokenized inputs and targets.",
+    )
+    parser.add_argument(
+        "--inputs_lid_col",
+        type=str,
+        default="inputs_lid",
+        help="Column name for inputs language ID.",
+    )
+    parser.add_argument(
+        "--targets_lid_col",
+        type=str,
+        default="targets_lid",
+        help="Column name for targets language ID.",
+    )
+    parser.add_argument(
+        "--dataset_split",
+        type=str,
+        default="train",
+        help="Dataset split to use (e.g., train, test, validation).",
+    )
+    parser.add_argument(
+        "--min_inputs_length",
+        type=int,
+        default=0,
+        help="Minimum length of the `inputs` field to keep the row.",
+    )
+    parser.add_argument(
+        "--min_targets_length",
+        type=int,
+        default=0,
+        help="Minimum length of the `targets` field to keep the row.",
+    )
+    parser.add_argument(
+        "--clean_samples_csv",
+        type=str,
+        default="lid_clean_samples.csv",
+        help="Filename for the output CSV with clean samples where LIDs match the assigned language.",
+    )
+    parser.add_argument(
+        "--duplicated_samples_csv", type=str, default="lid_duplicated_samples.csv"
+    )
+    parser.add_argument(
+        "--short_samples_csv", type=str, default="lid_short_samples.csv"
+    )
+    parser.add_argument(
+        "--wrong_language_csv",
+        type=str,
+        default="lid_wrong_language_samples.csv",
+        help="Filename for the output CSV with samples where both LIDs are wrong.",
+    )
+    parser.add_argument(
+        "--mismatch_samples_csv",
+        type=str,
+        default="lid_mismatch_samples.csv",
+        help="Filename for the output CSV with samples where either the input or target LID differs from the assigned language.",
+    )
+
+    args = parser.parse_args()
+
+    # Load the dataset
+    dataset = load_dataset(args.dataset_name)
+    df = pd.DataFrame(dataset[args.dataset_split])
+
+    # Load the tokenized CSV file
+    tokenized_df = pd.read_csv(args.tokenized_csv_filename)
+
+    # Load the lid CSV file
+    lid_df = pd.read_csv(args.lid_csv_filename)
+
+    # Merge dataset with LID results
+    m_df = pd.concat([df, lid_df, tokenized_df], axis=1)
+
+    # 1. Drop duplicates
+    duplicated_df = m_df[m_df.duplicated(subset=["inputs", "targets"], keep="first")]
+    duplicated_df.to_csv(args.duplicated_samples_csv, index=False)
+    print(f"Duplicated samples saved to {args.duplicated_samples_csv}")
+    m_df.drop_duplicates(subset=["inputs", "targets"], keep="first", inplace=True)
+
+    # 2. Filter out rows based on minimum length criteria
+    short_df = m_df[
+        (m_df["inputs_n_tokens_stanza"] < 4)
```

Comment:
@AMR-KELEG There is also the `args.min_inputs_length` and `args.min_targets_length`  that were introduced in this PR and are used on lines 115 to 120 -- I believe it would make sense to use them instead of hardcoding the 4s in here :)

Reply:
Oh yes, thanks for pointing this out.
These are number of tokens not number of characters, though.